staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -492,6 +492,26 @@ type Disk struct {
 	// If specified, disk address and its tag will be provided to the guest via config drive metadata
 	// +optional
 	Tag string `json:"tag,omitempty"`
+	// If specified, the virtual disk will be presented with the given block sizes.
+	// +optional
+	BlockSize *BlockSize `json:"blockSize,omitempty"`
+}
+
+// CustomBlockSize represents the desired logical and physical block size for a VM disk.
+//
+// +k8s:openapi-gen=true
+type CustomBlockSize struct {
+	LogicalBlockSize  uint `json:"logical,omitempty"`
+	PhysicalBlockSize uint `json:"physical,omitempty"`
```

Comment:
You are correct, only the top level should have been `omitempty`. I'll correct that. 
For the v1 side I'll rename the struct fields instead since you're already in a `blockSize:` field and that would be very redundant. I'll keep it as-is on the domain side though since that represents the libvirt blockio XML element.